### PR TITLE
#73 How to read this document

### PR DESCRIPTION
**Pull request by anatoly-scherbakov**
Introduce the `How to read this document` section and a stub for Primer section as per #73.

@juusoautiosalo fyi :) For some reason I cannot request review from you.

***
<a href="https://pr-preview.s3.amazonaws.com/json-ld/yaml-ld/pull/80.html" title="Last updated on Aug 29, 2022, 5:17 AM UTC (aa267fa)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/json-ld/yaml-ld/80/27be7af...aa267fa.html" title="Last updated on Aug 29, 2022, 5:17 AM UTC (aa267fa)">Diff</a>

### CODE REVIEW

File: spec/index.html [hunk 1]
```diff
@@ -298,6 +298,62 @@ <h2>Introduction</h2>
...
+            <li>
+              whereas RDF familiarity is only required when it is desired to convert YAML-LD to RDF graphs,
+              or vice versa.
+            </li>
+          </ul>
+        </li>
+        <li>
+          <p>
+            <strong>IT and non-IT professionals</strong> who want to read and/or produce Linked Data documents
+            in YAML-LD format. Such documents can be:
```
Comment (TallTed) [inline spec/index.html:339]:
```suggestion
            in YAML-LD format. Such documents can be &mdash;
```

File: spec/index.html [hunk 2]
```diff
@@ -298,6 +298,62 @@ <h2>Introduction</h2>
...
+
+            <ul>
+              <li>consumed by programming systems capable of understanding YAML-LD,</li>
+              <li>transformed with JSON-LD framing algorithms,</li>
+              <li>published on the Web for human and machine consumption.</li>
```
Comment (TallTed) [inline spec/index.html:344]:
```suggestion
              <li>consumed by programming systems capable of understanding YAML-LD</li>
              <li>transformed with JSON-LD framing algorithms</li>
              <li>published on the Web for human and machine consumption</li>
```

File: spec/index.html [hunk 3]
```diff
@@ -298,6 +298,62 @@ <h2>Introduction</h2>
...
+            </ul>
+          </p>
+          <p>
+            For these users, familiarity with JSON-LD is not required, but understanding of Linked Data principles
+            might be beneficial. <a href="#primer">YAML-LD Primer</a> is meant as a gentle introduction
```
Comment (TallTed) [inline spec/index.html:349]:
```suggestion
            might be beneficial. The <a href="#primer">YAML-LD Primer</a> is meant as a gentle introduction
```

File: spec/index.html [hunk 4]
```diff
@@ -1952,5 +2008,21 @@ <h2>Best Practices</h2>
...
+  <section id="primer" class="informative">
+    <h2>YAML-LD Primer</h2>
+
+    <section id="primer-introduction">
+      <h2>Introduction</h2>
+
+      <p>
+        This document is meant as a gentle introduction to YAML-LD for not just software developers and IT
+        professionals but also for the general audience of professionals in domains other than
+        computer science.
```
Comment (TallTed) [inline spec/index.html:2021]:
```suggestion
        This document is meant as a gentle introduction to YAML-LD for software developers and IT
        professionals as well as professionals in domains other than
        computer science.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by gkellogg** [comment]
> Introduce the `How to read this document` section and a stub for Primer section as per #73.
> 
> @juusoautiosalo fyi :) For some reason I cannot request review from you.

Juuso needs to be added as a contributor to the repository, which I can take care of.

File: spec/index.html [hunk 5]
```diff
@@ -298,6 +298,62 @@ <h2>Introduction</h2>
...
+        <li>and basic Linked Data [[LINKED-DATA]] principles.</li>
+      </ul>
+    </p>
+
+    <p>This document is intended primarily for the following two main audiences.</p>
+
+    <p>
+      <ul>
+        <li>
+          <strong>Software developers</strong> who want to:
```
[the remark this answers is not in the document]
Comment (anatoly-scherbakov) [reply spec/index.html:317]: @TallTed thank you for the review! But I am not exactly clear why you propose to replace `:` → `—` here? I was accustomed to having `:` in such places.

File: spec/index.html [hunk 6]
```diff
@@ -298,6 +298,62 @@ <h2>Introduction</h2>
...
+          <ul>
+            <li>encode Linked Data in a variety of programming languages which can use YAML,</li>
+            <li>convert existing YAML to YAML-LD,</li>
+            <li>understand the design decisions and language syntax for YAML-LD,</li>
+            <li>implement processors and APIs for YAML-LD,</li>
+            <li>generate or consume Linked Data, an RDF Graph, or an RDF Dataset in a YAML syntax.</li>
```
[the remark this answers is not in the document]
Comment (anatoly-scherbakov) [reply spec/index.html:323]: @TallTed the same question here: I was accustomed to see commas on such bullet lists.

Is there perhaps a W3C English style guide?

File: spec/index.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (gkellogg) [reply spec/index.html:317]: I have to say, as a native speaker, this is a bit mysterious to me, as well. There's a useful [writing guide](https://writingcenter.unc.edu/tips-and-tools/semi-colons-colons-and-dashes/) with the following statement:

> 1. To set off material for emphasis. Think of dashes as the opposite of parentheses. Where parentheses indicate that the reader should put less emphasis on the enclosed material, dashes indicate that the reader should pay more attention to the material between the dashes. ...

I'd certainly like to hear @TallTed's reasoning for these cases, though. I generally trust his sense of these things.

**Review by gkellogg — APPROVED** [review]
This is an important piece, thanks for adding. I support @TallTed's suggested changes. I am a bit suspect that non-IT professionals can effectively make use of this without also understanding the core principles of JSON-LD.

File: spec/index.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (TallTed) [reply spec/index.html:317]: Basically, colons should only follow full sentences, which "Software developers who want to" is not. Typically, rephrasing to include "the following" or similar allows the colon to be used — but "Software developers" is already the object of a preceding sentence, so this (as well as the other) would require additional, more complicated rephrasing. Ending a phrase (an incomplete sentence) that leads into bullets with an em-dash is generally accepted as an alternative.

File: spec/index.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (TallTed) [reply spec/index.html:323]: There is a [W3C manual of style](https://www.w3.org/Guide/manual-of-style/), but it's largely a handwave to each author'a/editor's own copy/ies of _The Chicago Manual of Style_ and/or _The Gregg Reference Manual_.  I can't cite my copies of these as they are in uncataloged boxes which have not yet been unpacked from my latest move.  I might not be keeping to the letter of these guides, either, as I have largely developed my editorial voice as "what works best for my audience," where that audience is the eventual readers of this document, not so much the other contributors thereto.

Generally speaking, I find that list items that are short (single words, small phrases) should not be terminally punctuated, and every item in that list should be kept short. Longer list items that contain full sentences may be terminally punctuated, but should then also be initially capitalized, and every item in that list should be a full sentence. Neither of these is iron-clad, but work well in general.

**Pull request merged**